File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
      minLength = float("infinity")
      start = 0
      windowSum = 0

      for end, num in enumerate(nums):
        windowSum += num

        while windowSum >= target:
          minLength = min(minLength, end - start + 1)
          windowSum -= nums[start]
          start += 1
        
      return minLength if minLength != float("infinity") else 0

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */

var minSubArrayLen = function(target, nums) {
  let start = 0;
  let minLength = Infinity;
  let windowSum = 0;

  for (let end = 0; end < nums.length; end++) {
    windowSum += nums[end];

    while (windowSum >= target) {
      minLength = Math.min(minLength, end - start + 1);
      windowSum -= nums[start];
      start++;
    }
  }

  if (minLength === Infinity) return 0;
  return minLength
};

